PresenceInsightsSDK/PresenceInsightsSDK/PIAdapter.swift

```diff
@@ -239,13 +252,17 @@ extension PIAdapter {
         
         let request = self.buildRequest(endpoint, method: self.PUT, body: deviceJSON)
         self.performRequest(request, callback: {response, error in
-            guard error == nil else {
-                callback(PIDevice(), error)
+            guard let response = response where error == nil else {
+                callback(nil, error)
                 return
             }
             self.printDebug("Update Response: \(response)")
+            // TODO: check if this test is at the right place
```

Comment:
Yes, this is a good place for this test. "@code" is a required field and if it doesn't exist then something went wrong.